itest/rust/src/dictionary_test.rs

```diff
@@ -400,3 +372,279 @@ fn dictionary_equal() {
     assert_eq!(dict! {1: f32::NAN}, dict! {1: f32::NAN}); // yes apparently godot considers these equal
     assert_ne!(dict! {"foo": "bar"}, dict! {"bar": "foo"});
 }
+
+#[itest]
+fn dictionary_iter_create() {
+    let dictionary = dict! {
+        "foo": 0,
+        "bar": true,
+        "baz": "foobar",
+        "nil": Variant::nil(),
+    };
+
+    dictionary.iter_shared();
+    dictionary.keys_shared();
+}
+
+#[itest]
+fn dictionary_iter() {
+    let dictionary = dict! {
+        "foo": 0,
+        "bar": true,
+        "baz": "foobar",
+        "nil": Variant::nil(),
+    };
+
+    let map = HashMap::<String, Variant>::from([
+        ("foo".into(), 0.to_variant()),
+        ("bar".into(), true.to_variant()),
+        ("baz".into(), "foobar".to_variant()),
+        ("nil".into(), Variant::nil()),
+    ]);
+    for (key, value) in dictionary.iter_shared().typed::<String, Variant>() {
+        assert!(map.contains_key(&key), "key = \"{key}\"");
+        assert_eq!(map[&key], value, "key = \"{key}\"");
+    }
+
+    let map2: HashMap<_, _> = dictionary
+        .iter_shared()
+        .typed::<String, Variant>()
+        .collect();
```

Comment:
Would type inference still work if you wrote this?
```rs
    let map2: HashMap<String, Variant> = dictionary
        .iter_shared()
        .typed()
        .collect();
```
I think it's slightly clearer, as you only need one `< >` pair and no turbofish :)

Reply:
i agree, and it does work (though i love turbofish :c)